src/acl_mem.cpp

```diff
@@ -1521,6 +1521,12 @@ ACL_EXPORT CL_API_ENTRY cl_mem CL_API_CALL clCreateImageIntelFPGA(
     BAIL(local_errcode_ret);
   }
 
+  if (image_desc == NULL) {
+    BAIL_INFO(CL_INVALID_IMAGE_DESCRIPTOR, context, "image_desc is NULL");
+  }
+
+  cl_image_desc _image_desc = *image_desc;
```

Comment:
I think it's better to use the guidelines in the [specification](https://registry.khronos.org/OpenCL/sdk/1.2/docs/man/xhtml/cl_image_desc.html), which are probably what the users are most familiar with. Since it is public, it makes more sense to adhere to them or else we will have to also make changes to the guidelines. Therefore, I also believe the original code is better. Thank you @pcolberg for the investigation! I'll read more of the kronos guidelines in the future.